File: examples/java/fulfillment/src/main/java/dev/angzarr/examples/fulfillment/FulfillmentState.java
package dev.angzarr.examples.fulfillment;

public record FulfillmentState(
    String orderId,
    String status,
    String trackingNumber,
    String carrier,
    String pickerId,
    String packerId,
    String signature
) {
    public static FulfillmentState empty() {
        return new FulfillmentState("", "", "", "", "", "", "");
    }

    public boolean exists() {
        return !orderId.isEmpty();
    }
}

File: examples/kotlin/fulfillment/src/main/kotlin/dev/angzarr/examples/fulfillment/FulfillmentState.kt
package dev.angzarr.examples.fulfillment

data class FulfillmentState(
    val orderId: String = "",
    val status: String = "",
    val trackingNumber: String = ""
) {
    fun exists(): Boolean = orderId.isNotEmpty()
    fun isPending(): Boolean = status == "pending"
    fun isPicking(): Boolean = status == "picking"
    fun isPacking(): Boolean = status == "packing"
    fun isShipped(): Boolean = status == "shipped"
    fun isDelivered(): Boolean = status == "delivered"

    companion object {
        fun empty(): FulfillmentState = FulfillmentState()
    }
}
